### PR TITLE
feat: ✨ implement getPreferences and getDefaultPostPrivacy for Firefish

### PR DESCRIPTION
**Pull request by ThatOneCalculator**
Good-enough for now implementation for Firefish `accounts/preferences`

### CODE REVIEW

**Comment by ThatOneCalculator** [comment]
I'm honestly unsure why this happens. I don't get this locally when linting...

image

**Comment by h3poteto** [comment]
It will be resolved by https://github.com/h3poteto/megalodon/pull/1962
Please rebase this PR with the master branch.

**Comment by ThatOneCalculator** [comment]
@h3poteto Thank you for fixing that issue :) All checks pass now

File: megalodon/src/firefish.ts [hunk 1]
```diff
@@ -953,12 +953,26 @@ export default class Firefish implements MegalodonInterface {
...
-      const err = new NoImplementedError('firefish does not support')
-      reject(err)
+    return this.client.post<FirefishAPI.Entity.UserDetailMe>('/api/i').then(async res => {
+      return Object.assign(res, {
+        data: FirefishAPI.Converter.userPreferences(res.data, await this.getDefaultPostPrivacy())
+      })
     })
   }
 
+  public async getDefaultPostPrivacy(): Promise<'public' | 'unlisted' | 'private' | 'direct'> {
```
Comment (h3poteto) [inline megalodon/src/firefish.ts:963]: This method is not defined in megalodon interface. If this method is used in only `getPereferences`, please change `public` to `private`.

Comment (ThatOneCalculator) [reply megalodon/src/firefish.ts]: Since the function is used directly in Firefish, I defined it in the Megalodon interface.

**Review by h3poteto — APPROVED** [review]
Thank you

**Pull request merged**